code/data_structures/linked_list/linked_lists_operations/insertionInALinkedList.cpp

```diff
@@ -0,0 +1,60 @@
+#include<iostream>
+using namespace std;
+
+// Eveerytime you work with Append or Push function, there is a corner case. That when a NULL node is passed you have to return a node that points to the given value. There are two ways you can do, if you receive the double pointer as an argument or if you are returning node.
+
+struct node{
+    int data;
+    struct node* next;
+};
+
+void printList(struct node *head){
+    while(head){
+        cout<<head->data<<" ";
+        head = head->next;
+    }
+    cout<<endl;
+}
+
+void push(struct node** headref, int x){
```

Comment:
Whenever you use node, there is no need to prefix it with struct, since this is C++, not C!